pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -4,6 +4,96 @@ Template for each `dtype` helper function for hashtable
 WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 """
 
+ctypedef fused table_t:
+    kh_float64_t
+    kh_uint64_t
+    kh_int64_t
+    kh_pymap_t
+
+
+cdef kh_init(table_t **table_ptr):
+    if table_t is kh_float64_t:
+        table_ptr[0] = kh_init_float64()
+    elif table_t is kh_int64_t:
+        table_ptr[0] = kh_init_int64()
+    elif table_t is kh_uint64_t:
+        table_ptr[0] = kh_init_uint64()
+    elif table_t is kh_pymap_t:
+        table_ptr[0] = kh_init_pymap()
+    else:
+        raise NotImplementedError
+
+
+cdef inline void kh_resize(table_t *table, int n) nogil:
+    if table_t is kh_float64_t:
+        kh_resize_float64(table, n)
+    if table_t is kh_uint64_t:
+        kh_resize_uint64(table, n)
+    if table_t is kh_int64_t:
+        kh_resize_int64(table, n)
+    if table_t is kh_pymap_t:
+        with gil:
+            kh_resize_pymap(table, n)
+
+
+cdef inline kh_destroy(table_t *table):
```

Comment:
Does the destructor need to be inline? We don't have a standard for when to inline vs not but this would seem like overkill?

Reply:
I don't think it does, no.  No strong opinion.